bundles/pcmjava/tools.vitruv.applications.pcmjava.common/src/tools/vitruv/applications/pcmjava/common/pcm2java/Pcm2JavaCommon.reactions

```diff
@@ -42,12 +42,23 @@ execute actions in Java
 // ###################################################
 // ################ PACKAGE REACTIONS ################
 
-reaction CreatedRepository {
-	after element pcm::Repository created and inserted as root
+reaction RenamedRepository {
+	after attribute replaced at pcm::Repository[entityName]
 	call {
-		val repository = newValue;
-		createJavaPackage(repository, null, repository.entityName, "repository_root");
-		createRepositorySubPackages(repository);
+		createRepositoryPackages(affectedEObject)
+		renamePackageForRepository(affectedEObject)
+	} 
+}
+
+routine createRepositoryPackages(pcm::Repository repository) {
+	match {
+		require absence of java::Package corresponding to repository
+	}
+	action {
+		call {
+			createJavaPackage(repository, null, repository.entityName, "repository_root");
+			createRepositorySubPackages(repository);
+		} 
 	}
 }
```

Comment:
I implemented the approach where the rename reaction creates the repository packages when they are absent. With this version there is an issue with the rename test case, where the Java packages for the default name repository (aName) are not properly removed. I will analyze where that comes from.

Reply:
It is indeed another issue that was previously not visible due to a different execution order of the change propagation specification. This means this PR can be merged!